test-integration/server/kb/structure/EdgeIT.java

```diff
@@ -71,26 +70,21 @@
     public void setUp(){
         String keyspaceName = "ksp_"+UUID.randomUUID().toString().substring(0, 20).replace("-", "_");
         Keyspace keyspace = new KeyspaceImpl(keyspaceName);
-        final int TIMEOUT_MINUTES_ATTRIBUTES_CACHE = 2;
-        final int ATTRIBUTES_CACHE_MAX_SIZE = 10000;
 
         // obtain components to create sessions and transactions
         JanusGraphFactory janusGraphFactory = server.janusGraphFactory();
         StandardJanusGraph graph = janusGraphFactory.openGraph(keyspace.name());
 
         // create the session
-        Cache<String, ConceptId> attributeCache = CacheBuilder.newBuilder()
-                .expireAfterAccess(TIMEOUT_MINUTES_ATTRIBUTES_CACHE, TimeUnit.MINUTES)
-                .maximumSize(ATTRIBUTES_CACHE_MAX_SIZE)
-                .build();
+        AttributeManager attributeManager = new AttributeManagerImpl();
 
         session = new SessionImpl(keyspace, server.serverConfig(), new KeyspaceSchemaCache(), graph,
-                new KeyspaceStatistics(), attributeCache, new ReentrantReadWriteLock());
+                new KeyspaceStatistics(), attributeManager, new ShardManagerImpl(), new ReentrantReadWriteLock());
 
         // create the transaction
         CacheProviderImpl cacheProvider = new CacheProviderImpl(new KeyspaceSchemaCache());
         UncomittedStatisticsDelta statisticsDelta = new UncomittedStatisticsDelta();
-        ConceptObserver conceptObserver = new ConceptObserver(cacheProvider, statisticsDelta);
+        ConceptObserver conceptObserver = new ConceptObserver(cacheProvider, statisticsDelta, attributeManager, graph.toString());
```

Comment:
graph.toString()?  shouldn't it be `txId`?

Reply:
graph.toString() is what we currently use as an Id. COuld revise that.

Reply:
https://github.com/graknlabs/grakn/pull/5492/files#diff-7226e2759816d4691e666d175f134c4aR157

here we are using transaction.toString()

Reply:
Ok true.